### ISSUE TITLE
Error - Maximum call stack size exceeded

### ISSUE THREAD

**Issue by pratikgaikwadhive** [javascript-obfuscator/javascript-obfuscator#961]
Attempting to obfuscate a file via node or cli. Results in same error. It used to work in version `1.8.0` of javascript-obfuscator. 

## Steps to Reproduce
```
./node_modules/.bin/javascript-obfuscator --compact true --source-map false --log true --target node dist/lib/utilities/fabricContext.js
```

## Your Environment

* Obfuscator version used: "2.17.0"
* Node version used: "v16.4.1"(Ubuntu 18.04 docker) and "v16.5.0"(OSX)

# Stack trace
```
root@d2f732800eaa:/workspaces/hive-sense# ./node_modules/.bin/javascript-obfuscator --compact true --source-map false --log true --target node dist/lib/utilities/fabricContext.js 

[javascript-obfuscator-cli] Obfuscating file: dist/lib/utilities/fabricContext.js... 

[javascript-obfuscator] Version: 2.17.0_2021-07-19T20:41:09.918Z

[javascript-obfuscator] Obfuscation started... 

[javascript-obfuscator] Random generator seed: 191302133...

[javascript-obfuscator] Code transformation stage: PreparingTransformers...

[javascript-obfuscator] AST transformation stage: Initializing...

[javascript-obfuscator] AST transformation stage: Preparing...

[javascript-obfuscator-cli] Error in file: dist/lib/utilities/fabricContext.js... 

/workspaces/hive-sense/node_modules/javascript-obfuscator/dist/webpack:/javascript-obfuscator/src/analyzers/scope-analyzer/ScopeAnalyzer.ts:71
                throw new Error(error);
                      ^
Error: RangeError: Maximum call stack size exceeded
    at o.analyze (/workspaces/hive-sense/node_modules/javascript-obfuscator/dist/webpack:/javascript-obfuscator/src/analyzers/scope-analyzer/ScopeAnalyzer.ts:71:23)
    at o.traverseScopeIdentifiers (/workspaces/hive-sense/node_modules/javascript-obfuscator/dist/webpack:/javascript-obfuscator/src/node/ScopeIdentifiersTraverser.ts:26:28)
    at _.transformNode (/workspaces/hive-sense/node_modules/javascript-obfuscator/dist/webpack:/javascript-obfuscator/src/node-transformers/preparing-transformers/VariablePreserveTransformer.ts:54:40)
    at visitorFunction (/workspaces/hive-sense/node_modules/javascript-obfuscator/dist/webpack:/javascript-obfuscator/src/node-transformers/preparing-transformers/VariablePreserveTransformer.ts:45:41)
    at Controller.<anonymous> (/workspaces/hive-sense/node_modules/javascript-obfuscator/dist/webpack:/javascript-obfuscator/src/node-transformers/NodeTransformersRunner.ts:108:39)
    at Controller.__execute (/workspaces/hive-sense/node_modules/@javascript-obfuscator/estraverse/estraverse.js:336:31)
    at Controller.replace (/workspaces/hive-sense/node_modules/@javascript-obfuscator/estraverse/estraverse.js:594:27)
    at Object.replace (/workspaces/hive-sense/node_modules/@javascript-obfuscator/estraverse/estraverse.js:675:27)
    at N.transform (/workspaces/hive-sense/node_modules/javascript-obfuscator/dist/webpack:/javascript-obfuscator/src/node-transformers/NodeTransformersRunner.ts:74:24)
    at o.runNodeTransformationStage (/workspaces/hive-sense/node_modules/javascript-obfuscator/dist/webpack:/javascript-obfuscator/src/JavaScriptObfuscator.ts:143:44)
```

## Minimal working example that will help to reproduce issue
Contents of `FabricContext.js`
```
"use strict";
class FabricContext {
    static _instance = null;
    _hostid = "";
    _clusterId = "";
    constructor() {
    }
    static get Instance() {
        return this._instance || (this._instance = new FabricContext());
    }
    set context(context) {
        this._hostid = context.hostid;
        this._clusterId = context.clusterId;
    }
    get hostid() {
        return this._hostid;
    }
    get clusterId() {
        return this._clusterId;
    }
}
module.exports = FabricContext.Instance;
```

**Comment by sanex3339** [comment]
Related with https://github.com/javascript-obfuscator/javascript-obfuscator/issues/909

**Comment by sanex3339** [comment]
TLDR: don't use class properties until the official obfuscator support

**Comment by sanex3339** [comment]
Released as `2.18.0`

**Comment by pratikgaikwadhive** [comment]
@sanex3339,

thank you for quick turn around. But I'm getting below error with `v2.18.0/1`
```
error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.

24         const buildDate: string = new Date(parseInt(buildTimestamp, 10)).toISOString();
                                                       ~~~~~~~~~~~~~~
```

**Comment by sanex3339** [comment]
Please update your dependencies
